### PR TITLE
Update scala example to latest fauna

### PR DESCRIPTION
**Pull request by fauna-arnaud**
This sample has been really useful so I decided to keep it updated with the latest improvements.

Commits can be reviewed individually.

### CODE REVIEW

**Review by trevorsibanda — APPROVED** [review]

**Review by evbruno — APPROVED** [review]

**Comment by fauna-arnaud** [comment]
@lregnier Can I merge this? :smiley:

File: scala/faunadb-crud-example-app/README.md [hunk 1]
```diff
@@ -85,6 +76,15 @@ Make sure to write down the given secret key. It will be used for starting up th
 
 > Alternatively, you can also create an API Key from the Cloud Dashboard [here](https://dashboard.fauna.com/db/demo-app/keys).
 
+
+### 5. Set up Schema
+
+Run below command for creating the DB schema. It will execute all required queries from a file.
+
+```
+$ fauna eval demo-app --file=./scripts/create_schema.fql --secret=xxxxxxx
```
[the remark this answers is not in the document]
Comment (lregnier) [reply scala/faunadb-crud-example-app/README.md:85]: Also, we mentioned the issued API secret as follows later on the README file:

```
your_api_key_goes_here
```

Maybe we could keep it consistent?  (either `your_api_key_goes_here` or `xxxxxxx` works fine for me!)

[the remark this answers is not in the document]
Comment (lregnier) [reply scala/faunadb-crud-example-app/README.md:85]: I've just checked the docs, and we're doing it as follows there:

```
'YOUR_FAUNADB_SERVER_SECRET'
```

https://docs.fauna.com/fauna/current/tutorials/crud?lang=javascript#clientserverkey

:)

File: scala/faunadb-crud-example-app/project/Versions.scala [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-  val scalaVersion              = "2.12.8"
+  val scalaVersion              = "2.12.12"
   val javaVersion               = "1.8"
 
   val akkaHttpVersion           = "10.1.5"
   val akkaHttpJson4sVersion     = "1.22.0"
   val json4sVersion             = "3.6.1"
 
-  val faunaDbVersion            = "2.6.0"
+  val faunaDbVersion            = "3.0.0"
```
Comment (lregnier) [inline scala/faunadb-crud-example-app/project/Versions.scala:9]: 🚀

File: scala/faunadb-crud-example-app/src/main/scala/persistence/FaunaRepository.scala [hunk 3]
```diff
@@ -146,13 +145,13 @@ trait FaunaRepository[A <: Entity] extends Repository[A] with FaunaRepository.Im
...
-    val beforeCursor = po.before.map(id => Before(Ref(Class(className), id)))
-    val afterCursor = po.after.map(id => After(Ref(Class(className), id)))
+    val beforeCursor = po.before.map(id => Before(Ref(Collection(collectionName), id)))
+    val afterCursor = po.after.map(id => After(Ref(Collection(collectionName), id)))
     val cursor = beforeCursor.orElse(afterCursor).getOrElse(NoCursor)
 
     val result = client.query(
       Map(
-        Paginate(Match(Index(classIndexName)), size = po.size, cursor = cursor),
+        Paginate(Documents(Collection(collectionName)), size = po.size, cursor = cursor),
```
Comment (lregnier) [inline scala/faunadb-crud-example-app/src/main/scala/persistence/FaunaRepository.scala:154]: 👏  Cool to see `Documents` here!

**Review by lregnier — APPROVED** [review]
Changes look good! Thank you for keeping the example up to date! 😃 💪

I've just added a small comment about the `fauna eval` shell command.

And also, check out the "FQL Reference" section in the README file. It seems there are some `Class("posts")` and one `Index("all_posts")` references that need to be updated there as well! :)

**Comment by fauna-arnaud** [comment]
@lregnier I have reviewed entirely the `README` and addressed all your comments I believe :+1:

**Pull request merged**